src/hooks/useOnPropsChange/index.ts

```diff
@@ -0,0 +1,3 @@
+import useOnPropsChange from './useOnPropsChange';
+
+export default useOnPropsChange;
```

Comment:
Do we need an index file for this? It makes sense to have one for `components` and `hooks` folders. But this will always just be one hook that we are exposing, or will it be more than one file for the other things we will have in here? and we want to keep a common pattern? If that makes sense

Reply:
This is the convention that it's better to be kept through the project (in order to prevent not having barrel index files in some hooks/components, have them in some other ones)

This hook is simple, but imagine a hook with some sort of useReduce for handeling the state of hook (which demands more complexity like, extra _enum_ types, utils functions and maybe extraction of the reduce part into separate file). 

In that case this index file will be the entry point to that hook and you will call that complex hook just by importing like below: 
```tsx
import myComplextHook from '@qlik-oss/nebula-table-utils/lib/hooks/myComplextHook'; // <- nothing extra than it's folder name
``` 
instead of reaching into exact file like this: 
```tsx
import myComplextHook from '@qlik-oss/nebula-table-utils/lib/hooks/myComplextHook/myComplextHook';
``` 

I agree with you that in this case its just one file, and because of that, it might seems like overkill, but in the same `myComplextHook` scenario, we can export types like: 
```tsx
import myComplextHook from './myComplextHook';

export default myComplextHook;
export * from "./types";
```
from the index file (barrel file) and have all of those types available for us in where we use them like:
```ts
import myComplextHook, { MyInterface, MyType } from '@qlik-oss/nebula-table-utils/lib/hooks/myComplextHook';

```

now think about components, they may have some other things also to share, and also if we integrate storybook for components, it will get handy for us when we want to do exports storybook files. 


So in general, it's a good practice for libraries to have such a thing.

Reply:
Didn't know this much about barrel files, great to learn. Then let's keep it as is, makes total sense.